ps3.py

```diff
@@ -0,0 +1,276 @@
+#
+# The 6.0001 Word Game
+# Created by: Kevin Luu <luuk> and Jenna Wiens <jwiens>
+#
+# Name          : Danilo Mizin
+# Collaborators :
+# Time spent    :
+
+import math
+import random
+import string
+import re
+
+VOWELS = 'aeiou'
+CONSONANTS = 'bcdfghjklmnpqrstvwxyz'
+HAND_SIZE = 7
+
+SCRABBLE_LETTER_VALUES = {
+    'a': 1, 'b': 3, 'c': 3, 'd': 2, 'e': 1, 'f': 4, 'g': 2, 'h': 4, 'i': 1, 'j': 8, 'k': 5, 'l': 1, 'm': 3, 'n': 1,
+    'o': 1, 'p': 3, 'q': 10, 'r': 1, 's': 1, 't': 1, 'u': 1, 'v': 4, 'w': 4, 'x': 8, 'y': 4, 'z': 10, "*": 0
+}
+
+WORDLIST_FILENAME = "words.txt"
+
+
+def load_words():
+    """
+    Returns a list of valid words. Words are strings of lowercase letters.
+
+    Depending on the size of the word list, this function may
+    take a while to finish.
+    """
+
+    print("Loading word list from file...")
+    # inFile: file
+    inFile = open(WORDLIST_FILENAME, 'r')
+    # wordlist: list of strings
+    wordlist = []
+    for line in inFile:
+        wordlist.append(line.strip().lower())
+    print("  ", len(wordlist), "words loaded.")
+    return wordlist
+
+
+def get_frequency_dict(sequence):
+    """
+    Returns a dictionary where the keys are elements of the sequence
+    and the values are integer counts, for the number of times that
+    an element is repeated in the sequence.
+    """
+
+    # freqs: dictionary (element_type -> int)
+    freq = {}
+    for x in sequence:
+        freq[x] = freq.get(x, 0) + 1
+    return freq
+
+
+def get_word_score(word, n):
+    """
+    Returns the score for a word. Assumes the word is a
+    valid word.
+    """
+    count = 0
+    for letter in word.lower():
+        count += SCRABBLE_LETTER_VALUES[letter]
+    return count * max(1, (7 * len(word) - 3 * (n - len(word))))
+
+
+def display_hand(hand):
+    """
+    Displays the letters currently in the hand.
+    """
+
+    for letter in hand.keys():
+        for j in range(hand[letter]):
+            print(letter, end=' ')  # print all on the same line
+    print()  # print an empty line
+
+
+def deal_hand(n):
+    """
+    Returns a random hand containing n lowercase letters.
+    ceil(n/3) letters in the hand should be VOWELS (note,
+    ceil(n/3) means the smallest integer not less than n/3).
+    """
+
+    hand = {}
+    num_vowels = int(math.ceil(n / 3))
+
+    for i in range(num_vowels):
+        x = random.choice(VOWELS)
+        hand[x] = hand.get(x, 0) + 1
+
+    for i in range(num_vowels, n):
+        x = random.choice(CONSONANTS)
+        hand[x] = hand.get(x, 0) + 1
+    hand["*"] = 1
+    for letter in hand:
+        if letter in VOWELS and hand[letter] == 1:
+            del hand[letter]
+            return hand
+
+
+def update_hand(hand, word):
+    """
+    Updates the hand: uses up the letters in the given word
+    and returns the new hand, without those letters in it.
+    """
+    new_hand = hand.copy()
+    for letter in word.lower():
+        if letter in new_hand and new_hand[letter] > 0:
```

Comment:
Use `new_hand.get(letter, 0)` instead of this. There is a chance that the letter in word will not be in hand, and Python will throw an error.